chain/client/src/client_actions.rs

```diff
@@ -1781,6 +1785,21 @@ impl ClientActionHandler<ReshardingResponse> for ClientActions {
     }
 }
 
+impl ClientActionHandler<LoadMemtrieResponse> for ClientActions {
+    type Result = ();
+
+    #[perf]
+    fn handle(&mut self, msg: LoadMemtrieResponse) -> Self::Result {
+        tracing::debug!(target: "client", ?msg);
+        if let Some((sync, _, _)) = self.client.catchup_state_syncs.get_mut(&msg.sync_hash) {
+            // We are doing catchup
+            sync.set_load_memtrie_result(msg.shard_id, msg.load_result);
+        } else {
+            self.client.state_sync.set_load_memtrie_result(msg.shard_id, msg.load_result);
```

Comment:
what does this branch represent? maybe also add a command for this branch as well or put a common comment before the if statement?